kube-derive/src/lib.rs

```diff
@@ -148,6 +148,9 @@ mod resource;
 /// Sets the description of the schema in the generated CRD. If not specified
 /// `Auto-generated derived type for {customResourceName} via CustomResource` will be used instead.
 ///
+/// ## `#[kube(label = r#"{ "key": "LABEL_KEY", "value": "LABEL_VALUE"}"#))]`
```

Comment:
If we are going to preprocess the values (mapping them into a btreemap), we should probably do explicit key value such as:

```rust
#[kube(label(key = key, value = value))]
```

which i see is possible because [serde does it](https://serde.rs/variant-attrs.html#rename). This way users can use variables (constants) to reference the keys/values (users hate writing these out manually in many places as they get out of sync).

---

Alternatively, if you could __maybe__ do:

```rust
#[kube(label = (key, value))]
```

as with that, our internal generation code can use [`BTreeMap::from`](https://doc.rust-lang.org/std/collections/struct.BTreeMap.html#impl-From%3C%5B(K,+V);+N%5D%3E-for-BTreeMap%3CK,+V%3E) (possibly with a String::from map first) rather than having to extract key, values.

but this would rely on being able to pass a tuple to the proc macro, which i am not sure if is possible.

Reply:
Rewrote the solution based on your feedback